From b70dc68da08588d292f221404571dde98cb38e41 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Thu, 4 May 2017 22:22:18 +0800
Subject: [PATCH 1/1] ktree: build as obj-y rather than lib-y

This patch is to fix potential exported symbols undefined because they might
be discarded due to no users in vmlinux when the only user is configured as
kernel module.

ERROR: "ktree_init" [drivers/soc/ti/knav_qmss_qos.ko] undefined!
ERROR: "ktree_put_node" [drivers/soc/ti/knav_qmss_qos.ko] undefined!
ERROR: "ktree_set_root" [drivers/soc/ti/knav_qmss_qos.ko] undefined!
ERROR: "ktree_sort_children" [drivers/soc/ti/knav_qmss_qos.ko] undefined!
ERROR: "ktree_add_child_before" [drivers/soc/ti/knav_qmss_qos.ko] undefined!
ERROR: "ktree_for_each_child" [drivers/soc/ti/knav_qmss_qos.ko] undefined!
ERROR: "ktree_remove_node" [drivers/soc/ti/knav_qmss_qos.ko] undefined!
scripts/Makefile.modpost:91: recipe for target '__modpost' failed
make[1]: *** [__modpost] Error 1

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 lib/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/Makefile b/lib/Makefile
index e307546..41a26e2 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -28,7 +28,7 @@ lib-$(CONFIG_MMU) += ioremap.o
 lib-$(CONFIG_SMP) += cpumask.o
 lib-$(CONFIG_HAS_DMA) += dma-noop.o
 
-lib-y	+= kobject.o klist.o ktree.o
+lib-y	+= kobject.o klist.o
 obj-y	+= lockref.o
 
 obj-y += bcd.o div64.o sort.o parser.o halfmd4.o debug_locks.o random32.o \
@@ -36,7 +36,7 @@ obj-y += bcd.o div64.o sort.o parser.o halfmd4.o debug_locks.o random32.o \
 	 gcd.o lcm.o list_sort.o uuid.o flex_array.o iov_iter.o clz_ctz.o \
 	 bsearch.o find_bit.o llist.o memweight.o kfifo.o \
 	 percpu-refcount.o percpu_ida.o rhashtable.o reciprocal_div.o \
-	 once.o
+	 once.o ktree.o
 obj-y += string_helpers.o
 obj-$(CONFIG_TEST_STRING_HELPERS) += test-string_helpers.o
 obj-y += hexdump.o
-- 
2.7.4

